src/Compilers/CSharp/Portable/Lowering/LocalRewriter/LocalRewriter_Patterns.cs

```diff
@@ -1,164 +1,581 @@
+                            if (!decisionResult)
+                            {
+                                _conjunctBuilder.Add(_factory.Literal(decisionResult));
+                                return true;
+                            }
+                        }
+                        else if (d.Value == ConstantValue.Null)
+                        {
+                            addConjunct(ref this, _localRewriter.MakeNullCheck(d.Syntax, input, input.Type.IsNullableType() ? BinaryOperatorKind.NullableNullEqual : BinaryOperatorKind.Equal));
+                        }
+                        else
+                        {
+                            var systemObject = _factory.SpecialType(SpecialType.System_Object);
+                            addConjunct(ref this, _localRewriter.MakeEqual(_factory.Literal(d.Value, input.Type), input));
+                        }
+                        break;
+                    case BoundDagFieldEvaluation f:
+                        {
+                            var field = f.Field;
+                            var outputTemp = new BoundDagTemp(f.Syntax, field.Type, f, 0);
+                            var output = _tempAllocator.GetTemp(outputTemp);
+                            _sideEffectBuilder.Add(_factory.AssignmentExpression(output, _factory.Field(input, field)));
+                        }
+                        break;
+                    case BoundDagPropertyEvaluation p:
+                        {
+                            var property = p.Property;
+                            var outputTemp = new BoundDagTemp(p.Syntax, property.Type, p, 0);
+                            var output = _tempAllocator.GetTemp(outputTemp);
+                            _sideEffectBuilder.Add(_factory.AssignmentExpression(output, _factory.Property(input, property)));
+                        }
+                        break;
+                    case BoundDagDeconstructEvaluation d:
+                        {
+                            var method = d.DeconstructMethod;
+                            var refKindBuilder = ArrayBuilder<RefKind>.GetInstance();
+                            var argBuilder = ArrayBuilder<BoundExpression>.GetInstance();
+                            BoundExpression receiver;
+                            void addArg(RefKind refKind, BoundExpression expression)
+                            {
+                                refKindBuilder.Add(refKind);
+                                argBuilder.Add(expression);
+                            }
+                            Debug.Assert(method.Name == "Deconstruct");
+                            int extensionExtra;
+                            if (method.IsStatic)
+                            {
+                                Debug.Assert(method.IsExtensionMethod);
+                                receiver = _factory.Type(method.ContainingType);
+                                addArg(RefKind.None, input);
```

Comment:
>RefKind.None [](start = 39, length = 12)

Is it safe to assume that the ref kind is None here? I think the code will be more robust if we simply pull value from the corresponding parameter.

Reply:
Resolved in [https://github.com/dotnet/roslyn/pull/23357/commits/758fd76932a6d9bcb640504c8fdef57fb5adecec](https://github.com/dotnet/roslyn/pull/23357/commits/758fd76932a6d9bcb640504c8fdef57fb5adecec) of [https://github.com/dotnet/roslyn/pull/23357](https://github.com/dotnet/roslyn/pull/23357)

---
In reply to: [156560306](https://github.com/dotnet/roslyn/pull/23209#discussion_r156560306) [](ancestors = 156560306)